test/test_make_report.py

```diff
@@ -5,17 +5,46 @@
 class TestAltText(unittest.TestCase):
```

Comment:
You could actually parametrize these using something like this (I've never tried it this way though): https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests

I've always used `pytest` rather than the built in `unittest` because it has much more functionality (but requires installing pytest - add it to `requirements.txt` then run `pip install -r requirements.txt` again whilst inside the venv) - then you'd do it like this: https://docs.pytest.org/en/6.2.x/parametrize.html. The `ids` argument is optional, it just gives a name to each parametrized test which gives slightly nicer error messages when it fails.
```python
@pytest.mark.parametrize("filename, expected_alt_text", [
     ("0_ben_doing_comething_silly.jpg", "Ben doing something silly"),
     ("0_0_john_does_something.jpg", "John does something"),
     ...etc
], ids=["one_numeric_prefix", "two_numeric_prefixes", ...etc])
def test_get_image_caption(filename: str, expected_alt_text: str):
    alt_text = get_image_caption(filename)
    assert alt_text == expected_alt_text
```

If you want to leave it as it is that's fine though - let me know and I'll merge it!